extensions/amp-video/0.1/amp-video.js

```diff
@@ -259,11 +260,20 @@ export class AmpVideo extends AMP.BaseElement {
     // Cached so mediapool operations (eg: swapping sources) don't interfere with this bool.
     this.hasBitrateSources_ =
       !!this.element.querySelector('source[data-bitrate]') ||
-      this.hasAnyCachedSources_();
+      this.hasAnyCachedSources_() ||
+      this.element.hasAttribute('enable-google-video-cache');
 
     installVideoManagerForDoc(element);
 
     Services.videoManagerForDoc(element).register(this);
+
+    // Add cache sources if opted-in and it's not a cached doc already.
```

Comment:
Being a cached doc !== having cached sources, your `if` is right but please update the comment